var/spack/repos/builtin/packages/lammps/package.py

```diff
@@ -133,45 +155,80 @@ def url_for_version(self, version):
     depends_on('libpng', when='+png')
     depends_on('ffmpeg', when='+ffmpeg')
     depends_on('kokkos+deprecated_code+shared@3.0.00', when='@20200303+kokkos')
     depends_on('kokkos+shared@3.1:', when='@20200505:+kokkos')
     depends_on('adios2', when='+user-adios')
+    depends_on('adios2', when='+adios')
     depends_on('plumed', when='+user-plumed')
+    depends_on('plumed', when='+plumed')
     depends_on('eigen@3:', when='+user-smd')
+    depends_on('eigen@3:', when='+machdyn')
 
     conflicts('+cuda', when='+opencl')
     conflicts('+body', when='+poems@:20180628')
     conflicts('+latte', when='@:20170921')
     conflicts('+python', when='~lib')
     conflicts('+qeq', when='~manybody')
     conflicts('+user-atc', when='~manybody')
+    conflicts('+atc', when='~manybody')
     conflicts('+user-misc', when='~manybody')
     conflicts('+user-phonon', when='~kspace')
-    conflicts('+user-misc', when='~manybody')
+    conflicts('+phonon', when='~kspace')
     conflicts('%gcc@9:', when='@:20200303+openmp')
     conflicts('+kokkos', when='@:20200227')
     conflicts(
-        '+meam', when='@20181212:',
-        msg='+meam was removed after @20181212, use +user-meamc instead')
+        '+meam', when='@20190104:20210527',
+        msg='+meam was removed from version @20190104 to version @20210527, ' +
+            'use +user-meamc instead')
     conflicts(
-        '+user-meamc', when='@:20181212',
-        msg='+user-meamc only added @20181212, use +meam instead')
+        '+user-meamc', when='@:20181212,20210702:',
+        msg='+user-meamc only supported after version @20181212 and before ' +
+            'version @20210702, use +meam instead')
     conflicts(
         '+user-reaction', when='@:20200303',
         msg='+user-reaction only supported for version 20200505 and later')
     conflicts('+mliap', when='~snap')
+    conflicts('+ml-iap', when='~ml-snap')
     conflicts(
         '+user-adios +mpi', when='^adios2~mpi',
         msg='With +user-adios, mpi setting for adios2 and lammps must be the same')
     conflicts(
         '+user-adios ~mpi', when='^adios2+mpi',
         msg='With +user-adios, mpi setting for adios2 and lammps must be the same')
+    conflicts(
+        '+adios +mpi', when='^adios2~mpi',
+        msg='With +adios, mpi setting for adios2 and lammps must be the same')
+    conflicts(
+        '+adios ~mpi', when='^adios2+mpi',
+        msg='With +adios, mpi setting for adios2 and lammps must be the same')
+
+    for old_pkg, new_pkg in renamed_packages_20210702.items():
+        conflicts(
+            '+{0}'.format(old_pkg), when='@20210702:',
+            msg='+{0} was renamed in version @20210702, '.format(old_pkg) +
+                'use +{0} instead'.format(new_pkg))
+        conflicts(
+            '+{0}'.format(new_pkg), when='@:20210527',
+            msg='+{0} was introduced in version @20210702, '.format(new_pkg) +
+                'use +{0} instead'.format(old_pkg))
```

Comment:
Wouldn't using the new [conditional variant feature](https://spack.readthedocs.io/en/latest/packaging_guide.html#conditional-variants) be more suitable for this.

Reply:
@rashidrafeek : I didn't think about it but I guess it makes sense to use that feature. The only drawback is that you cannot control the error message so you cannot redirect the user to the right variant.

@adamjstewart @alalazo : What would you recommend in this case?

Reply:
I prefer to use conditional variants for features added or removed in certain versions and conflicts for features broken on certain versions or platforms.

Reply:
Maybe the name of the renamed variant can be included in the variant description? Like "[var] was renamed in [version]. Use [renamed-var] for newer versions." And vice-versa for the renamed variants. 

But the current approach also seems fine.

Reply:
I can try this and see if it's still readable.

Reply:
Why not rename the variant for all versions?

Reply:
No sure to be honest, maybe @junghans @akohlmey could comment on this suggestion?

Reply:
@RemiLacroix-IDRIS I would rename the variants for all versions and then go from there. We should also deprecate all versions prior to at least 2021. Let me know if you need some help with this.

Reply:
Rename as in give the variants the same names as the current packages' names and internally handle the change of package name for older versions?

I suppose that means keeping the old variants' names for when the packages have been split? (Merged packages can be handled automatically I guess.)

Reply:
yes, only keep the new names and use old conditional variant names for options that truly went away.

Reply:
@rbberger : Ok, I will try to implement that suggestion and see how it goes.